languages/csharp/csharp.go

```diff
@@ -47,10 +48,38 @@ func (l language) Imports(content []byte, root *sitter.Node) ([]string, error) {
 		}
 
 		for _, cap := range m.Captures {
-			str := string(content[cap.Node.StartByte():cap.Node.EndByte()])
-			out = append(out, str)
+			out = append(out, filterOut(content, cap.Node))
 		}
 	}
 
 	return out, nil
 }
+
+func filterOut(content []byte, node *sitter.Node) string {
+	var (
+		filters = map[string]struct{}{
+			"type_argument_list": struct{}{},
+		}
+
+		out string
+
+		fn func(n *sitter.Node)
+	)
+
+	fn = func(n *sitter.Node) {
+		if _, ok := filters[n.Type()]; ok {
+			return
+		}
+
+		if cnt := int(n.ChildCount()); cnt == 0 {
```

Comment:
It's pretty hard for a reader to verify logic here.

Is it intended to match on a specific token type only? Or all tokens?

I propose to check the type here instead. I'm pretty sure you only need a single one (`identifier`?), and then you can `Join` by `.` to get the same result. It would be _a lot_ more readable.

Reply:
To be more specific, try this case:
```c#
using f = Foo.Bar /* comment */ <int> ;
```

Reply:
Please, take a look.
It should be simpler - we have _include_ and _exclude_ filters, because some identifiers can be nested children, e.g. `using mystack = Namespace.Stack<MyType>`